python/ray/experimental/signal.py

```diff
@@ -0,0 +1,184 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import hashlib
+import time
+
+import ray
+
+START_SIGNAL_COUNTER = 10000
+
+class Signal(object):
+    """Signal object"""
+    pass
+
+class DoneSignal(Signal):
+    pass
+
+class ErrorSignal(Signal):
+    def __init__(self, error):
+        self.error = error
+
+
+def _get_signal_id(source_id, counter):
+    if  type(source_id) is ray.actor.ActorHandle:
+        return ray._raylet.compute_signal_id(
+            ray._raylet.compute_task_id(source_id._ray_actor_creation_dummy_object_id),
+            counter)
+    else:
+        # TODO(pcm): Figure out why compute_signal_id is called with both
```

Comment:
This needs to be fixed now that we are more strict about distinguishing TaskIDs and ObjectIDs.